### PR TITLE
 cephadm: require podman >= 2.0.0

### PR DESCRIPTION
**Pull request by mgfritch**
Fixes: https://tracker.ceph.com/issues/47139
Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mgfritch** [comment]
Tested on both ubuntu bionic, openSUSE tumbleweed, and CentOS 8.0.

Most cephadm commands will require podman >= 2.0:
```
root@ubuntu:/home/ubuntu# ./cephadm ls
ERROR: podman version 2.0.0 or later is required
```

check-host will also verify the version:
```
root@ubuntu:/home/ubuntu# ./cephadm check-host
systemctl is present
lvcreate is present
Unit systemd-timesyncd.service is enabled and running
ERROR: podman version 2.0.0 or later is required
```

A subsequent run of `cephadm prepare-host` should pull in the newer version of podman.

**Comment by smithfarm** [comment]
In `ceph.spec.in` we have 

```
Recommends:      podman
```

and in `debian/control` we have

```
Recommends: podman | docker.io
```

Would it be a good idea for this PR to change that to:

```
Recommends:    podman >= 2
```

and

```
Recommends: podman (>= 2) | docker.io
```

?

**Comment by smithfarm** [comment]
RATIONALE: It seems weird for RPM to install podman as a dependency, only for cephadm to refuse to work with it because the podman version is too old.

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -11,6 +11,7 @@ UNIT_DIR = '/etc/systemd/system'
 LOG_DIR_MODE = 0o770
 DATA_DIR_MODE = 0o700
 CONTAINER_PREFERENCE = ['podman', 'docker']  # prefer podman to docker
+MIN_PODMAN_VERSION = (2, 0, 0)
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:14]: 2.0.2 . AFAIK 2.0.1 and 2.0.0 were broken.

**Review by sebastian-philipp — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-01-28_09:47:15-rados:cephadm-wip-swagner3-testing-2021-01-27-1346-distro-basic-smithi/

* https://tracker.ceph.com/issues/48981

I'm super surprised that the run passed.

**Pull request merged**